lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -257,12 +257,24 @@ def get_code_string(self):
 """
         return result
 
+    def _write_query_plan_to_trace_dir(self, plan: LogicalPlan, query_id: str):
+        assert self.trace_dir is not None, "Writing query_plan requires trace_dir to be set"
+        path = os.path.join(self.trace_dir, f"{query_id}/metadata/")
+        os.makedirs(path, exist_ok=True)
+        with open(os.path.join(path, "query_plan.json"), "w") as f:
+            f.write(plan.json())
```

Comment:
I think you want `model_dump_json()` here, right? `json()` was for Pydantic v1 (it still exists but I think it is deprecated).